### ISSUE TITLE
Compare to holographic graph embeddings

### ISSUE THREAD

**Issue by alexanderpanchenko** [uhh-lt/path2vec#7]
Report results on our datasets for the following models in this order:

- **TransX** (all models starting from "Trans" from [2]).

- **HolE** (holographic embeddings from [1]).

[1] holographic:
https://github.com/laxatives/GraphEmbeddings

[2] TransX:
https://github.com/thunlp/TensorFlow-TransX

**Comment by alexanderpanchenko** [comment]
@alexteua we are preparing a new version of this paper for the NAACL conference and need your further help. Could you please compute another baseline on all datasets for this paper, in the similar way as you have done it with the Deepwalk, for the model linked above? This is OK to start to do in beginning of October  - we still have some time (this will relate to the hiwi thing).

**Comment by alexteua** [comment]
@alexanderpanchenko sure, I will start with this in the beginning of October

**Comment by alexteua** [comment]
Instead of [2], I used pure C++ implementation from the same repository
https://github.com/thunlp/Fast-TransX
I run it on WordNet data and did some sanity check for resulting embeddings.

**Comment by alexteua** [comment]
Here some examples:

similar to:  cauliflower.n.02

DeepWalk
--------------
kohlrabi.n.02 0.9882519245147705
broccoli_rabe.n.02 0.9881056547164917
brussels_sprouts.n.01 0.9867178797721863
mustard.n.03 0.9865701198577881
broccoli.n.02 0.9838613271713257
cruciferous_vegetable.n.01 0.9731203317642212
radish.n.01 0.9082521200180054
turnip.n.02 0.8662794828414917
white_turnip.n.02 0.8651506304740906
rutabaga.n.01 0.8647150993347168

transD
--------------
broccoli.n.02 0.83204585313797
brussels_sprouts.n.01 0.7510706186294556
rutabaga.n.01 0.6474558115005493
broccoli_rabe.n.02 0.625973641872406
mustard.n.03 0.6190798878669739
kohlrabi.n.02 0.6169585585594177
radish.n.01 0.5720855593681335
ligature.n.04 0.5386346578598022
disk_overhead.n.01 0.5247583389282227
tetralogy_of_fallot.n.01 0.5092943906784058

similar to:  entree.n.04

DeepWalk
--------------
registration.n.01 0.9880821704864502
irruption.n.01 0.9861668944358826
incursion.n.01 0.9860333204269409
intrusion.n.02 0.9858594536781311
entrance.n.03 0.9633533954620361
penetration.n.03 0.9575451612472534
admission.n.01 0.9505788087844849
matriculation.n.01 0.9356059432029724
interpenetration.n.02 0.9255791902542114
readmission.n.01 0.9150411486625671

transD
--------------
registration.n.01 0.6589906215667725
intrusion.n.02 0.6531375646591187
incursion.n.01 0.6482845544815063
irruption.n.01 0.616395115852356
sternness.n.01 0.5650294423103333
territoriality.n.01 0.5023216009140015
galley.n.03 0.4939059615135193
angler.n.02 0.4777261018753052
neckcloth.n.01 0.4762835204601288
turkish_bath.n.01 0.4754646420478821

**Comment by alexteua** [comment]
scores for the similar words for transD could be lower because of small number of epochs I did for this first try. But it seems meaningful.

**Comment by alexteua** [comment]
Now I will run other transX models with more epochs, and will do evaluation we used for another baseline methods. I will  post link to the google doc with first results here a bit later

**Comment by alexanderpanchenko** [comment]
Thanks, looks great. Looking forward to see the numbers - it is really interesting to know how these compare to deepwalk and to our model.

**Comment by alexteua** [comment]
I've generated vectors with different parameters and run evaluation. @akutuzov can you please give me access to the document you gave the link? I will check if table2 format is different from the table2 in the published paper and will upload results to the google doc

**Comment by akutuzov** [comment]
@alexteua Can you access the paper by this link? https://www.overleaf.com/read/zpgxymqgzrjp

Also, can you upload the models you trained to somewhere where I can reach them?

**Comment by alexteua** [comment]
@akutuzov  Yes, now it's accessible, thanks.
Just to clarify:
Is for table 2 this evaluation data used https://github.com/uhh-lt/path2vec/tree/master/simlex ?
And for table 3 https://github.com/uhh-lt/path2vec/tree/master/simlex/simlex_synsets ?
Should I perform evaluation on both evaluation sets?

**Comment by akutuzov** [comment]
@alexteua Yes, exactly (the tables are a bit in a mess now, because I'm adding new metrics).

**Comment by alexteua** [comment]
Great! I will upload results and embeddings to access a bit later today

**Comment by alexteua** [comment]
results for 4 transX models are available at transX sheet in google doc. please let me know if there are other combinations of parameters we need to test. 
trained embeddings are currently at ltcpu3, path is mentioned in google doc.

**Comment by alexteua** [comment]
results for holographic embeddings will be next soon

**Comment by alexanderpanchenko** [comment]
thanks a lot, great job!

**Comment by akutuzov** [comment]
Thanks @alexteua !
It seems that `epochs` and `learning rate` columns in the googledoc are mixed up. Can you check and fix that?

**Comment by alexteua** [comment]
@akutuzov thanks for noticing, fixed

**Comment by alexteua** [comment]
holographic embeddings are decided to skip
WSD evaluation for transX models is available in google doc

**Comment by alexanderpanchenko** [comment]
@alexteua please add to the table 

https://docs.google.com/spreadsheets/d/1KjNns16ld3pVUY1K7aA0HH9Lrb1PiKBH7-ZIbWrD9Kc/edit#gid=1895788589

evaluations on two other WSD datasets (currently you was using Senseval2, but we also need Senseval3 and SemEval 2015). Both new datasets are in exact same format and are available here:  http://lcl.uniroma1.it/wsdeval/evaluation-data. you just need to indicate path to the right dataset in the evaluation script. (This task is quite urgent).

**Comment by alexteua** [comment]
ok, will do

**Comment by alexanderpanchenko** [comment]
@alexteua please also add a short text description of the TransR, TransD and other two TransX methods to the overleaf in section 5.2 - see the fixme there in the baselines section.

You need to shortly describe the models (in the same style as other models are presented in this section) and shortly describe their differences.

**Comment by alexteua** [comment]
results for evaluation on senseval3 and semeval-2015 are added to the table

**Comment by alexteua** [comment]
will add description of the models to the paper as a next step

**Comment by alexanderpanchenko** [comment]
Thanks!

**Comment by alexanderpanchenko** [comment]
@akutuzov could you please integrate the result into the paper?

**Comment by akutuzov** [comment]
@alexanderpanchenko Done.

**Comment by alexteua** [comment]
I updated Related work section with description of 4 TransX models. Can you please check the size and content of that fragment?

**Comment by alexteua** [comment]
also I will add proper citations in that fragment

**Comment by akutuzov** [comment]
@alexteua thanks!
I think that this is even **too** detailed and we eventually will have to leave only the description of TransR and TransD. @alexanderpanchenko what do you think?

**Comment by alexteua** [comment]
@akutuzov I see, thanks. Also I can shorten it and we can keep 4 models

**Comment by alexanderpanchenko** [comment]
I will shorten. However I could not understand the description of the first model. Could you please specify more strictly what the “projection” there means? A random reader may not grasp this.

**Comment by alexteua** [comment]
yes, sure
